main.go

```diff
@@ -824,6 +831,10 @@ func main() {
              </body>
              </html>`))
 	})
-	log.Infoln("Starting HTTP server on", *listenAddress)
-	log.Fatal(http.ListenAndServe(*listenAddress, nil))
+
+	level.Info(logger).Log("msg", "Listening on address", "address", *listenAddress)
+	if err := http.ListenAndServe(*listenAddress, nil); err != nil {
+		level.Error(logger).Log("msg", "Error starting HTTP server", "err", err)
```

Comment:
```suggestion
		level.Error(logger).Log("msg", "Error running HTTP server", "err", err)
```

The error can happen a while after the startup as well.